src/shell.js

```diff
@@ -309,6 +309,9 @@ if (ENVIRONMENT_IS_SHELL) {
 
   if (typeof quit == 'function') {
     quit_ = (status, toThrow) => {
+      if (runtimeKeepaliveCounter) {
+        throw toThrow;
```

Comment:
`EXIT_RUNTIME=0` should mean the runtime stays alive and we just throw to unwind.   The fact that out test suite depends on the returns codes, but also doesn't set EXIT_RUNTIME=1 seems like maybe the problem here.  We work around it node using `process['exitCode'] = status;` but we can't extend that hack here I guess

Reply:
Yes, makes sense. I added a comment now to explain the differences between shell and node here, hopefully that is correct.